### PR TITLE
T36848 Set `Node.result` of timed out nodes and skipped tests

### PR DESCRIPTION
**Pull request by JenySadadia**
- fstests: update node result of skipped tests
- src/notifier: add `skip` and `incomplete` results
- src/timeout: set timed out node result
Implement method `_set_node_result` to set node result of timed-out nodes. If the node is in 'running' state, set the
node result to 'incomplete'.
When the timed-out node is not in 'running' state, set the node result to 'fail' if one of the child nodes failed. Otherwise, set the result to 'pass'.

### CODE REVIEW

File: src/timeout.py [hunk 1]
```diff
@@ -29,6 +29,26 @@ def __init__(self, configs, args):
...
+            node['result'] = 'incomplete'
+            return node
+        failed_child_nodes = self._db.count_nodes({
+            'parent': node['_id'],
+            'result': 'fail'
+        })
+        if failed_child_nodes:
+            node['result'] = 'fail'
+        else:
+            node['result'] = 'pass'
```
Comment (mgalka) [inline src/timeout.py:49]:
```suggestion
        node['result'] = 'fail' if  failed_child_nodes else 'pass'
```

Comment (JenySadadia) [reply src/timeout.py]: Updated.

**Comment by JenySadadia** [comment]
Tested OK on staging:
```
today at 13:35:1811/01/2022 08:05:18 AM UTC [INFO] Listening for events... 
today at 13:35:1811/01/2022 08:05:18 AM UTC [INFO] Press Ctrl-C to stop.
today at 13:35:1811/01/2022 08:05:18 AM UTC [INFO] Time                        Commit        Node Id                   State      Result    Name
today at 13:35:2111/01/2022 08:05:21 AM UTC [INFO] 2022-11-01 08:05:21.145876  2b1e56ae482c  6360d3418f94e20c6826b0b4  Running    -         checkout
today at 13:36:4511/01/2022 08:06:45 AM UTC [INFO] 2022-11-01 08:06:44.958396  bc6a17d622a0  635f81de207ffd6cb9d2caa4  Done       Incomplete  checkout
```

**Comment by JenySadadia** [comment]
Parent node results are being set successfully in case of failed child nodes:

```
$ curl 'https://staging.kernelci.org:9000/nodes?result=fail&name=checkout&limit=1'
{"items":[{"_id":"6352c322207ffd6cb9d2c0da","kind":"node","name":"checkout","path":["checkout"],"group":null,"revision":{"tree":"kernelci","url":"https://github.com/kernelci/linux.git","branch":"staging-stable","commit":"091d7ca79258ee626925bb7890f9198bfb1907f5","describe":"staging-stable-20221021.0","version":{"version":5,"patchlevel":15,"sublevel":null,"extra":"-10866-g091d7ca79258","name":null}},"parent":null,"state":"done","result":"fail","artifacts":{"tarball":"http://staging.kernelci.org:9080/linux-kernelci-staging-stable-staging-stable-20221021.0.tar.gz"},"created":"2022-10-21T16:04:50.610000","updated":"2022-10-21T16:16:43.834000","timeout":"2022-10-22T16:04:50.610000","holdoff":"2022-10-21T16:16:28.154000"}],"total":4,"limit":1,"offset":0}
```

```
$ curl 'https://staging.kernelci.org:9000/nodes?result=fail&parent=6352c322207ffd6cb9d2c0da'
{"items":[{"_id":"6352c384207ffd6cb9d2c0db","kind":"node","name":"kver","path":["checkout","kver"],"group":"kver","revision":{"tree":"kernelci","url":"https://github.com/kernelci/linux.git","branch":"staging-stable","commit":"091d7ca79258ee626925bb7890f9198bfb1907f5","describe":"staging-stable-20221021.0","version":{"version":5,"patchlevel":15,"sublevel":null,"extra":"-10866-g091d7ca79258","name":null}},"parent":"6352c322207ffd6cb9d2c0da","state":"done","result":"fail","artifacts":{"tarball":"http://staging.kernelci.org:9080/linux-kernelci-staging-stable-staging-stable-20221021.0.tar.gz"},"created":"2022-10-21T16:06:28.290000","updated":"2022-10-21T16:06:54.130000","timeout":"2022-10-22T16:06:28.290000","holdoff":null}],"total":1,"limit":50,"offset":0}
```

**Review by mgalka — APPROVED** [review]

**Pull request merged**